src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputList) {
+        inputList.remove(0);
+        return inputList.stream()
+                 .map(s -> s.split(","))
+                 .map(s -> new FruitTransaction(FruitTransaction.Operation.getFromChar(s[0]),

```

Comment:
According to the checklist, we should handle the Purchase operation by checking the result balance before saving it in the Storage. This conversion method should ensure that the quantity for a purchase is not resulting in a negative balance.